vllm/v1/engine/parallel_sampling.py

```diff
@@ -0,0 +1,181 @@
+# SPDX-License-Identifier: Apache-2.0
+
+from copy import copy
+from typing import Optional
+
+from vllm.outputs import RequestOutput
+from vllm.sampling_params import RequestOutputKind, SamplingParams
+
+
+class ParentRequestState:
+    """Info and state for parallel sampling request.
+    
+    Store parent request ID and sampling params.
+    Facilitate generating child request sampling params.
+    When stream mode is disabled, then `self.request_output`
+    aggregates completions.
+    """
+
+    request_id: str
+    sampling_params: SamplingParams
+    request_output: Optional[RequestOutput] = None
+
+    def __init__(self, request_id: str,
+                 sampling_params: SamplingParams) -> None:
+        self.request_id = request_id
+        self.sampling_params = sampling_params
+
+    def get_warmup_sampling_params(self, ) -> SamplingParams:
+        sampling_params = copy(self.sampling_params)
+        sampling_params.max_tokens = 1
+        sampling_params.n = 1
+        sampling_params.output_kind = RequestOutputKind.FINAL_ONLY
+        return sampling_params
+
+    def get_child_sampling_params(
+        self,
+        seed: Optional[int],
+    ) -> SamplingParams:
+        sampling_params = copy(self.sampling_params)
+        sampling_params.n = 1
+        sampling_params.seed = seed
+        return sampling_params
+
+    def add_output(
+        self,
+        child_req_output: RequestOutput,
+    ) -> None:
+        """Aggregate a parallel sampling child
+        request output.
+        
+        Non-stream-mode (`output_kind == FINAL_ONLY`) 
+        only. Inject correct parent request ID and
+        completion index.
+
+        Args:
+          child_req_output: a single request output
+                            from a parallel sampling
+                            child request.       
+        """
+        if self.request_output is None:
+            # Save the first request output; reinstate
+            # original request ID; metrics are not
+            # supported for parallel sampling
+            child_req_output.request_id = self.request_id
+            child_req_output.metrics = None
+            self.request_output = child_req_output
+        else:
+            # Aggregate additional completion into request
+            # output
+            new_completion = child_req_output.outputs[0]
+            new_completion.index = self.num_completions
+            self.request_output.outputs.append(new_completion)
```

Comment:
I think we should pass the index into this method and do:
```suggestion
            new_completion.index = index
            self.request_output.outputs[index] = new_completion
```

Of course will need to modify the if body above to to initialize the list to n elements and move the first completion to the correct index.

This is to ensure that the order will be deterministic when using seed.